lib/pbench/server/__init__.py

```diff
@@ -38,32 +38,24 @@ def __init__(self, cfg_name):
         super().__init__(cfg_name)
 
         # Now fetch some default common pbench settings that are required.
-        try:
-            self.TOP = Path(self.conf.get("pbench-server", "pbench-top-dir"))
-            if not self.TOP.is_dir():
-                raise BadConfig(f"Bad TOP={self.TOP}")
-            self.TMP = Path(self.conf.get("pbench-server", "pbench-tmp-dir"))
-            if not self.TMP.is_dir():
-                raise BadConfig(f"Bad TMP={self.TMP}")
-            self.LOGSDIR = Path(self.conf.get("pbench-server", "pbench-logs-dir"))
-            if not self.LOGSDIR.is_dir():
-                raise BadConfig(f"Bad LOGSDIR={self.LOGSDIR}")
-            self.BINDIR = Path(self.conf.get("pbench-server", "script-dir"))
-            if not self.BINDIR.is_dir():
-                raise BadConfig(f"Bad BINDIR={self.BINDIR}")
-            self.LIBDIR = Path(self.conf.get("pbench-server", "lib-dir"))
-            if not self.LIBDIR.is_dir():
-                raise BadConfig(f"Bad LIBDIR={self.LIBDIR}")
-            # the scripts may use this to send status messages
-            self.mail_recipients = self.conf.get("pbench-server", "mailto")
-            self.ARCHIVE = Path(self.conf.get("pbench-server", "pbench-archive-dir"))
-        except (NoOptionError, NoSectionError) as exc:
-            raise BadConfig(str(exc))
-        else:
-            self.INCOMING = self.TOP / "public_html" / "incoming"
-            # this is where the symlink forest is going to go
-            self.RESULTS = self.TOP / "public_html" / "results"
-            self.USERS = self.TOP / "public_html" / "users"
+        self.TOP = self._get_valid_dir_option("TOP", "pbench-server", "pbench-top-dir")
+        self.TMP = self._get_valid_dir_option("TMP", "pbench-server", "pbench-tmp-dir")
+        self.LOGSDIR = self._get_valid_dir_option(
+            "LOGSDIR", "pbench-server", "pbench-logs-dir"
+        )
+        self.BINDIR = self._get_valid_dir_option(
+            "BINDIR", "pbench-server", "script-dir"
+        )
+        self.LIBDIR = self._get_valid_dir_option("LIBDIR", "pbench-server", "lib-dir")
+        self.mail_recipients = self.conf.get("pbench-server", "mailto")
+        self.ARCHIVE = self._get_valid_dir_option(
+            "ARCHIVE", "pbench-server", "pbench-archive-dir"
+        )
+
+        self.INCOMING = self.TOP / "public_html" / "incoming"
```

Comment:
We can't use get_valid_dir_option() here because we don't necessarily have a logger yet?

Reply:
because we need self.TOP for this option